### ISSUE TITLE
renameFolder request with ~250 files in the folder keep on being cancelled and retrying

### ISSUE THREAD

**Issue by matthieu-foucault** [mozilla/thimble.mozilla.org#1803]
I have a folder with 255 svg images in it (country flags from [this github repo](https://github.com/lipis/flag-icon-css/tree/master/flags/4x3)) and when I try to rename the folder, the corresponding XHR request keeps on being cancelled or returns with an error 504.

The requests are still being sent after the folder is deleted, and reloading the project does not seem to improve the situation.

**Comment by humphd** [comment]
Interesting.  It sounds like we need to throw away renames if a delete action is inserted in the sync queue--something I thought we already did.   Probably when we add items delete operations to the `PathCache` we should look through the list to see if any items can get purged:

https://github.com/mozilla/thimble.mozilla.org/blob/fa3dde088dc016a562520e62a9df1f5558ba8f70/public/editor/scripts/project/project.js#L128

@gideonthomas does that sound right?

**Comment by gideonthomas** [comment]
@humphd, so I'm not sure how this is happening, maybe there's a bug in how we're prioritizing operations. But, from what I can see in https://github.com/mozilla/thimble.mozilla.org/blob/fa3dde088dc016a562520e62a9df1f5558ba8f70/public/editor/scripts/editor/js/fc/sync-manager.js#L276, we always prioritize folder renames (since they don't involve any local fs operations when doing the sync). So we run the folder rename first and then we prioritize file delete operations. So, the deletes should always happen after the rename. We also make sure that the paths in the queue that are already there for update/delete are updated to new paths when the folder rename happens. When delete operations are queued up, we resolve update operations in the queue. So, I'm not sure where things are going wrong.

The 504 error is interesting. I wonder if this has anything to do with AWS and our databases going down (or if the timing coincides)

**Comment by humphd** [comment]
Agreed, I'm surprised to hear this is happening.  We should try to duplicate now that s3 is back up, using the images @matthieu-foucault mentions above.
